### ISSUE TITLE
randomx program not showing

### ISSUE THREAD

**Issue by Gingeropolous** [moneroexamples/onion-monero-blockchain-explorer#190]
I used the randomx branch with the recent monero master.

everything works fine except for the display of the randomx program

https://testnet.xmrchain.net/randomx/1309514

is there a flag or something I'm missing?

**Comment by moneroexamples** [comment]
Will look into.

**Comment by moneroexamples** [comment]
The issue is that `rx_vm` in https://github.com/monero-project/monero/blob/master/src/crypto/rx-slow-hash.c#L65 is `static` now. This make it private variable,  thus I could relay on monero's randomx virtual machine like before.

**Comment by Gingeropolous** [comment]
oh well. It would have been cool to see it. 

If you want to go through the effort, you could probably re-create the program just using the input values. but that could be a lot of work for this.

**Comment by moneroexamples** [comment]
Yes I started doing it, but then other things needed for creating my own vm also become static, e,g, `rx_initdata` https://github.com/monero-project/monero/blob/master/src/crypto/rx-slow-hash.c#L65 Basically I think I would have to fork-off `rx-slow-hash.c` (or copy paste large chanks of it) and keep it as part the explorer. Not sure if doing this is worth maintaining it forever after that. 

Still looking into it, so still trying to figure something out.

**Comment by moneroexamples** [comment]
I forked `rx-slow-hash.c` and made the rx code work again. The changes are in this branch: https://github.com/moneroexamples/onion-monero-blockchain-explorer/tree/fix_randomx_code for now

You can also preview how it works on my testned node, e.g. http://139.162.60.17:8083/randomx/1315380
